sw/device/lib/crypto/include/kdf.h

```diff
@@ -19,22 +20,41 @@ extern "C" {
 #endif  // __cplusplus
 
 /**
- * Enum to define the supported KDF constructions.
+ * Performs the key derivation function in counter mode wtih HMAC according to
+ * NIST SP 800-108r1.
  *
- * Values are hardened.
+ * The supported PRF engine for the KDF function is HMAC (since KMAC does not
+ * use the counter mode).
+ *
+ * The caller should allocate and partially populate the `keying_material`
+ * blinded key struct, including populating the key configuration and
+ * allocating space for the keyblob. The caller should indicate the length of
+ * the allocated keyblob; this function will return an error if the keyblob
+ * length does not match expectations. For hardware-backed keys, the keyblob
+ * length is 0 and the keyblob pointer may be `NULL`. For non-hardware-backed
+ * keys, the keyblob should be twice the length of the key. The value in the
+ * `checksum` field of the blinded key struct will be populated by this
+ * function.
+ *
+ * @param key_derivation_key Pointer to the blinded key derivation key.
```

Comment:
For `key_derivation_key` it says "Pointer" in the comment, but below the parameter isn't a pointer type.
Same for `keying_material`.